packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts

```diff
@@ -169,9 +226,7 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
  * write to the transformer conf if the resolverConfig is valid
  */
 export const writeResolverConfig = async (conflictResolution: ConflictResolution, resourceDir) => {
-  if (conflictResolution && (conflictResolution.defaultResolutionStrategy || conflictResolution.perModelResolutionStrategy)) {
```

Comment:
Any reason why we removed this condition? Is this taken care in the "writeTransformerConfiguration" function?

Reply:
it's because this logic is now used for adding and updating and when updating you could remove all of the resolverConfigs in which case you need to write the config even if it's empty

Reply:
Okay, just make sure to check the following use case in the CLI:
Customer enables datastore when doing an add
Customer goes through the update flow but say "No, I don't want to update additional configurations".  
What happens in that case? Will the datastore configs be removed from the transform file or retained?

Reply:
Yeah that works. And actually, do we expect customers to be able to remove all of the transformer config entries or just change them? The walkthrough doesn't really make it possible

Reply:
Yeah, customers should be able to disable datastore if they want to. Do we have an option to do that in the headless options for update?